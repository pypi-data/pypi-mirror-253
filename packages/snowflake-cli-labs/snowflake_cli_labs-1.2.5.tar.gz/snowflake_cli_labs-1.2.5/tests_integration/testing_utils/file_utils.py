def replace_text_in_file(file_path: str, to_replace: str, replacement: str) -> None:
    with open(file_path, "r") as file:
        text = file.read()
    text = text.replace(to_replace, replacement)
    with open(file_path, "w") as file:
        file.write(text)
