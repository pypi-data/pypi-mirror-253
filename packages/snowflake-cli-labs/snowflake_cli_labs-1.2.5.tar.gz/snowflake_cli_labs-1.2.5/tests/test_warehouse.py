from tests.testing_utils.fixtures import *


@mock.patch("snowflake.connector")
def test_show_warehouses(mock_connector, mock_cursor, runner, snapshot):
    mock_connector.connect.return_value.execute_string.return_value = (
        None,
        mock_cursor(
            rows=[
                (
                    "foo",
                    "suspended",
                ),
                ("bar", "running"),
            ],
            columns=["name", "state"],
        ),
    )
    result = runner.invoke(["warehouse", "status"], catch_exceptions=False)
    assert result.exit_code == 0, result.output
    assert result.output == snapshot
