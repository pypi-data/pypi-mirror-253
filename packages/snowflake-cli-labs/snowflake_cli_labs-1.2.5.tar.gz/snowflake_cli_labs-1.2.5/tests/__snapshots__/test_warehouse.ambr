# serializer version: 1
# name: test_show_warehouses
  '''
  SELECT A MOCK QUERY
  +------------------+
  | name | state     |
  |------+-----------|
  | foo  | suspended |
  | bar  | running   |
  +------------------+
  
  '''
# ---
